lib/spack/spack/cmd/info.py

```diff
@@ -155,6 +155,32 @@ def print_text_info(pkg):
         color.cprint('')
         color.cprint(section_title('Maintainers: ') + mnt)
 
+    color.cprint('')
+    color.cprint(section_title('Externally Findable: '))
+    find_method = ''
+    is_findable = False
+    if hasattr(pkg, 'executables'):
+        is_findable = True
+        find_method += 'executables'
```

Comment:
"executables" is really a different type of information than "variants/versions/custom".  I think we should

1. Change the name to `attributes` instead of `method`
2. not include "executables" in this information (we don't detect the executables, we use them to detect various spec attributes)
3. check for an attribute 'executables' and only do the whole thing if that's set

Then if we later implement detection from libraries or pkgconf files, we can add that to the check for whether it's detectable. I think that a user typing `spack info foo` doesn't really care that we detect foo from an executable, what they care about is "can we find foo and what can we find out about it".